lib/atomic_cache/atomic_cache_client.rb

```diff
@@ -119,13 +119,11 @@ def last_known_value(keyspace, options, tags)
           return lkv
         end
 
-        # if the value of the last known key is nil, we can infer that it's
-        # most likely expired, thus remove it so other processes don't waste
-        # time trying to read it
-        @storage.delete(lkk)
```

Comment:
Dropping delete here, as we don't yet know if this is 100% safe. Added a metric in its place.